### PR TITLE
fix: downloadMediaMessage return type matched parameter

### PR DESCRIPTION
**Pull request by arthursimas1**

### CODE REVIEW

File: src/Utils/messages.ts [hunk 1]
```diff
@@ -868,12 +869,24 @@ const REUPLOAD_REQUIRED_STATUS = [410, 404]
 /**
  * Downloads the given message. Throws an error if it's not a media message
  */
-export const downloadMediaMessage = async(
```
Comment (azudindaem) [inline src/Utils/messages.ts:871]: can be simplified

```ts
export async function downloadMediaMessage(
  message: WAMessage,
  type: 'buffer' | 'stream',
  options: MediaDownloadOptions,
  ctx?: DownloadMediaMessageContext
): Promise<Buffer | Transform> {
  try {
    const result = await downloadMsg()
    return result
  } catch(error) {
    if(ctx && axios.isAxiosError(error) && REUPLOAD_REQUIRED_STATUS.includes(error.response?.status!)) {
      ctx.logger.info({ key: message.key }, 'sending reupload media request...')
      message = await ctx.reuploadRequest(message)
      return downloadMsg()
    }
    throw error
  }
```

Comment (arthursimas1) [reply src/Utils/messages.ts:871]: i'm sorry. i don't think so. in the way i've typed, the return type depends on the parameters. in that way:
- if `type = 'buffer'`, returns `Promise<Buffer>`
- if `type = 'stream'`, returns `Promise<Transform>`

**Review by smartsolucoesgo — APPROVED** [review]

File: src/Utils/messages.ts (hunk 1 above)
Comment (purpshell) [reply src/Utils/messages.ts:871]: could be simplified with `download<here>()`

Comment (purpshell) [reply src/Utils/messages.ts:871]:
```ts
export async function downloadMediaMessage<T extends 'buffer' | 'stream'>(
    message: WAMessage,
    type: T,
    options: MediaDownloadOptions,
    ctx?: DownloadMediaMessageContext
): Promise<T extends 'buffer' ? Buffer : Transform> {
   //
}
```

Comment (arthursimas1) [reply src/Utils/messages.ts:871]: good idea to use generics with extends! thank you
i wasn't able to make it exactly in that way (_typescript typing gymnastics_), but it's simpler than before

**Review by azudindaem — APPROVED** [review]

**Review by qikerrs — APPROVED** [review]

**Review by PurpShell — APPROVED** [review]

**Comment by arthursimas1** [comment]
@bobslavtriev use `{}` instead of passing nothing to in memory storage

File: src/Store/make-in-memory-store.ts [hunk 2]
```diff
@@ -29,7 +29,7 @@ export const waLabelAssociationKey: Comparable<LabelAssociation, string> = {
 export type BaileysInMemoryStoreConfig = {
 	chatKey?: Comparable<Chat, string>
 	labelAssociationKey?: Comparable<LabelAssociation, string>
-	logger: Logger
+	logger?: Logger
```
Comment (purpshell) [inline src/Store/make-in-memory-store.ts]:
```suggestion
	logger: Logger
```

File: src/Store/make-in-memory-store.ts [hunk 3]
```diff
@@ -74,12 +74,11 @@ const predefinedLabels = Object.freeze<Record<string, Label>>({
 })
 
 export default (
-	{ logger, chatKey, labelAssociationKey, socket }: BaileysInMemoryStoreConfig
+	{ logger: _logger, chatKey, labelAssociationKey, socket }: BaileysInMemoryStoreConfig
```
Comment (purpshell) [inline src/Store/make-in-memory-store.ts]:
```suggestion
	{ logger, chatKey, labelAssociationKey, socket }: BaileysInMemoryStoreConfig
```

File: src/Store/make-in-memory-store.ts [hunk 4]
```diff
@@ -74,12 +74,11 @@ const predefinedLabels = Object.freeze<Record<string, Label>>({
...
 ) => {
-	// const logger = _logger || DEFAULT_CONNECTION_CONFIG.logger.child({ stream: 'in-mem-store' })
+	const logger = _logger || DEFAULT_CONNECTION_CONFIG.logger.child({ stream: 'in-mem-store' })
```
Comment (purpshell) [inline src/Store/make-in-memory-store.ts]:
```suggestion
	// const logger = _logger || DEFAULT_CONNECTION_CONFIG.logger.child({ stream: 'in-mem-store' })
```

File: src/Store/make-in-memory-store.ts [hunk 5]
```diff
@@ -74,12 +74,11 @@ const predefinedLabels = Object.freeze<Record<string, Label>>({
...
 	chatKey = chatKey || waChatKey(true)
 	labelAssociationKey = labelAssociationKey || waLabelAssociationKey
-	logger = logger || DEFAULT_CONNECTION_CONFIG.logger.child({ stream: 'in-mem-store' })
 	const KeyedDB = require('@adiwajshing/keyed-db').default
```
Comment (purpshell) [inline src/Store/make-in-memory-store.ts]:
```suggestion
	logger = logger || DEFAULT_CONNECTION_CONFIG.logger.child({ stream: 'in-mem-store' })
	const KeyedDB = require('@adiwajshing/keyed-db').default
```

**Review by PurpShell — CHANGES_REQUESTED** [review]
remove

**Pull request merged**